packages/angular/projects/clr-angular/src/data/tree-view/recursive-for-of.ts

```diff
@@ -41,7 +42,7 @@ export class ClrRecursiveForOf<T> implements OnChanges, OnDestroy {
       wrapped = [new RecursiveTreeNodeModel(this.nodes, null, this.getChildren, this.featuresService)];
     }
     if (!this.childrenFetchSubscription) {
-      this.childrenFetchSubscription = this.featuresService.childrenFetched.subscribe(() => {
+      this.childrenFetchSubscription = this.featuresService.childrenFetched.pipe(debounceTime(0)).subscribe(() => {
```

Comment:
Can you please explain what are you achieving with `debounceTime`?

Reply:
Sure!
`RecursiveTreeModel#fetchChildren` is calling `childrenFetched.next()`
This happen on every single recursive tree node, defined like this:
```
 <clr-tree-node
      *clrRecursiveFor="let node of tree.root; getChildren: synchronousChildren"
```
This means that in a tree with 500 nodes on 1st level, we get 500 simultaneous change detection cycles.
The debounceTime here will help. run only 1 change detection for all simultaneous events coming from the getChildren calls.